File: src/main/java/com/youdeyiwu/enums/forum/PostStateEnum.java
package com.youdeyiwu.enums.forum;

import lombok.Getter;

/**
 * post state.
 * If you need to add an enumeration, please append it at the end.
 * This is because the enumeration is stored in natural order, not by name.
 *
 * @author dafengzhen
 */
@Getter
public enum PostStateEnum {

  /**
   * show (Default).
   */
  SHOW,

  /**
   * hide (Forum Administrator, Access to Section Management).
   */
  HIDE,

  /**
   * lock (The forum administrators, section moderators,
   * as well as the thread authors and users with access keys, can access).
   */
  LOCK,

  /**
   * allow (The forum administrators, section moderators,
   * as well as the thread authors and whitelisted users can access).
   */
  ALLOW,

  /**
   * block (Forum administrators, section administrators,
   * and post authors have access, while users on the blacklist cannot access).
   */
  BLOCK,

  /**
   * visible after login.
   */
  VISIBLE_AFTER_LOGIN

}

File: app/lib/enums/post_state_enum.dart
import 'package:json_annotation/json_annotation.dart';

enum PostStateEnum {
  /// show (Default).
  @JsonValue('SHOW')
  show,

  /// hide (Forum Administrator, Access to Section Management).
  @JsonValue('HIDE')
  hide,

  /// lock (The forum administrators, section moderators,
  /// as well as the thread authors and users with access keys, can access).
  @JsonValue('LOCK')
  lock,

  /// allow (The forum administrators, section moderators,
  /// as well as the thread authors and whitelisted users can access).
  @JsonValue('ALLOW')
  allow,

  /// block (Forum administrators, section administrators,
  /// and post authors have access, while users on the blacklist cannot access).
  @JsonValue('BLOCK')
  block,

  /// visible after login.
  @JsonValue('VISIBLE_AFTER_LOGIN')
  visibleAfterLogin
}
